packages/devtools_app/lib/src/memory/flutter/memory_chart.dart

```diff
@@ -80,17 +86,54 @@ class MemoryChartState extends State<MemoryChart> with AutoDisposeMixin {
       setState(() {
         controller.updatedMemorySource();
 
-        // Plot all offline or online data (based on memorySource).  If
-        // not offline then all new heap samples will be plotted as it
-        // appears via the sampleAddedNotifier.
-        _processAndUpdate(true);
+        // Reset all plotted data sets (Entries).
+        controller.memoryTimeline.chartData.reset();
+        controller.memoryTimeline.engineChartData.reset();
+
+        if (controller.offline) {
+          _recomputeOfflineData();
+          controller.memoryTimeline.chartData.reset();
+          controller.memoryTimeline.engineChartData.reset();
+          _processAndUpdate();
+        } else {
+          // Recompute data to display.
+          _recomputeChartData();
+
+          // Plot all offline or online data (based on memorySource).  If
+          // not offline then all new heap samples will be plotted as it
+          // appears via the sampleAddedNotifier and update all charts.
+          _processAndUpdate();
```

Comment:
_processAndUpdate is the last statement in both the if and else case.

@pq not flutter specific but have we thought about some general duplicate code lints/quick fixes like this?

Reply:
Duplicate code detection is something @bwilkerson has done some work on.  Maybe nudges like these will encourage him to get it landed... 😄

Reply:
Done.